### PR TITLE
 mdns: Initial version based on IDF 5.0 

### PR DESCRIPTION
**Pull request by gabsuren**

### CODE REVIEW

File: docs/docs_versions.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+var DOCUMENTATION_VERSIONS = {
+    DEFAULTS: { has_targets: false,
+                supported_targets: [ "esp32" ]
+              },
+    VERSIONS: [
+        { name: "latest", has_targets: true, supported_targets: [ "esp32", "esp32s2", "esp32s3", "esp32c2", "esp32c3" ] },
+
+        { name: "v4.4.1", old:false, has_targets:true, supported_targets: [ "esp32", "esp32s2", "esp32s3", "esp32c3" ]},
```
Comment (david-cermak) [inline docs/docs_versions.js:8]: I thought this version file describes the actual component, not IDF.

File: .github/workflows/test_afl_fuzzer.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Test AFL fuzzer host (mDNS)
```
Comment (david-cermak) [inline .github/workflows/test_afl_fuzzer.yml:1]: This only checks that the mdns fuzzer layers compile, correct? We shouldn't call it `AFL fuzzer`...

**Review by david-cermak — CHANGES_REQUESTED** [review]
LGTM in general. There are some minor point that I'm not convinced about:
1) GitHub action files (file names and names of actions and sub-actions) and their structure
2) Structure of tests -- `test_apps`, `test`, `host_test` in component root
3) Tests are not implemented/executed in the CI
4) `common_components` folder for the connection utility

But this could be addressed in future PRs...

File: common_components/protocol_examples_common/Kconfig.projbuild [hunk 3]
```diff
@@ -0,0 +1,322 @@
+menu "Example Connection Configuration"
```
[the remark this answers is not in the document]
Comment (gabsuren) [reply common_components/protocol_examples_common/Kconfig.projbuild:1]: https://github.com/espressif/esp-protocols/pull/26#discussion_r842677518

File: components/mdns/docs/en/index.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        find_mdns_service("_ftp", "_tcp"); //FTP server
+        //or networked printer
+        find_mdns_service("_printer", "_tcp");
+        find_mdns_service("_ipp", "_tcp");
+    }
+
+Execution Speed
+^^^^^^^^^^^^^^^
+
+ - mDNS creates a task with default low priority 1 `CONFIG_MDNS_TASK_PRIORITY` (If `CONFIG_FREERTOS_UNICORE` enabeled it pinned to CPU0 (`CONFIG_MDNS_TASK_AFFINITY`).
```
Comment (david-cermak) [inline components/mdns/docs/en/index.rst:182]:
```suggestion
 - mDNS creates a task with default low priority 1 `CONFIG_MDNS_TASK_PRIORITY` (If ``CONFIG_FREERTOS_UNICORE`` enabeled it pinned to CPU0 (`CONFIG_MDNS_TASK_AFFINITY`).
```

**Comment by gabsuren** [comment]
@david-cermak thanks for the review, updated accordingly and created new task to implement tests in CI
[IDFGH-7525](https://jira.espressif.com:8443/browse/IDFGH-7525)

**Review by david-cermak — APPROVED** [review]
Thanks for the latest updates. LGTM!

**Pull request merged**